cli/ballerina-cli-module/src/main/java/org/ballerinalang/cli/module/Pull.java

```diff
@@ -0,0 +1,227 @@
+                } else {
+                    createError("module version could not be detected");
+                }
+            }
+        } catch (IOException e) {
+            createError("connection to the remote repository host failed: " + e.getMessage());
+        }
+        conn.disconnect();
+        Authenticator.setDefault(null);
+    }
+
+    private static void createNightlyBuildMetaFile(Path nightlyBuildMetaFile) {
+        try {
+            Files.createFile(nightlyBuildMetaFile);
+        } catch (Exception e) {
+            createError("error occurred while creating nightly.build file.");
+        }
+    }
+
+    private static void copyBaloFile(String modulePath, HttpURLConnection conn, String moduleVersion, Path baloPath) {
+        try (InputStream inputStream = conn.getInputStream();
+                FileOutputStream outputStream = new FileOutputStream(baloPath.toString())) {
+            byte[] buffer = new byte[1024];
+            int count;
+
+            long totalSizeInKB = conn.getContentLengthLong() / 1024;
+            try (ProgressBar progressBar = new ProgressBar(
+                    modulePath + ":" + moduleVersion + " [central.ballerina.io -> home repo] ",
+                    totalSizeInKB, 1000, outStream, ProgressBarStyle.ASCII, " KB", 1)) {
+                while ((count = inputStream.read(buffer)) > 0) {
+                    outputStream.write(buffer, 0, count);
+                    progressBar.step();
+                }
+            }
+        } catch (IOException e) {
+            createError("error occurred copying the balo file: " + e.getMessage());
+        }
+    }
+
+    private static void createBaloFileDirectory(Path baloCacheWithModulePath) {
+        try {
+            Files.createDirectory(baloCacheWithModulePath);
+        } catch (Exception e) {
+            createError("error creating directory for balo file");
+        }
+    }
+
+    private static void handleErrorResponse(String url, String modulePath, boolean isBuild, HttpURLConnection conn,
+            int statusCode) {
+        try (BufferedReader reader = new BufferedReader(
+                new InputStreamReader(conn.getInputStream(), Charset.defaultCharset()))) {
+            StringBuilder result = new StringBuilder();
+            String line;
+            while ((line = reader.readLine()) != null) {
+                result.append(line);
+            }
+
+            MapValue payload = (MapValue) JSONParser.parse(result.toString());
+            String errMsg = payload.getStringValue("message");
+
+            if (statusCode == HttpURLConnection.HTTP_NOT_FOUND && isBuild && errMsg
+                    .contains("module not found")) {
+                // To ignore printing the error
+                createError("");
+            } else {
+                createError(errMsg);
+            }
+        } catch (IOException e) {
+            createError("failed to pull the module '" + modulePath + "' from the remote repository '" + url
```

Comment:
Shall we end the statement with the concatenation operator?

Reply:
Can you further explain this?

Reply:
```suggestion
            createError("failed to pull the module '" + modulePath + "' from the remote repository '" + url +
                                 "'");
```